l10n_it_pec_messages/model/mail_thread.py

```diff
@@ -154,9 +154,19 @@ def message_parse(
                 # I'm going to set this message as notification of the original
                 # message and remove the message_id of this message
                 # (it would be duplicated)
-                context['main_message_id'] = msg_ids[0]
-                context['pec_type'] = daticert_dict.get('pec_type')
-            del msg_dict['message_id']
+                # before deletion check if this message is prensent and linked with
+                # main massage id, if false remove message_id else no
```

Comment:
this line of domain search ('pec_msg_id', '=', daticert_dict['pec_msg_id']) make the test in comment
the field pec_msg_id is filled , when mail.message is an notification or delivery message,with message_id of parent message